NDTensors/src/lib/Sectors/src/symmetry_style.jl

```diff
@@ -0,0 +1,28 @@
+# This file defines SymmetryStyle, a trait to distinguish abelian groups, non-abelian groups
+# and non-group fusion categories.
+
+using ..LabelledNumbers: LabelledInteger, label
+
+abstract type SymmetryStyle end
+
+struct AbelianGroup <: SymmetryStyle end
+struct NonAbelianGroup <: SymmetryStyle end
+struct NonGroupCategory <: SymmetryStyle end
+struct EmptyCategory <: SymmetryStyle end  # CategoryProduct with zero category inside
```

Comment:
Is a better name for this `TrivialCategory`?

Reply:
@emstoudenmire curious to hear your thoughts on this.

Reply:
I understand "trivial category" as the category containing one object (similarly to the trivial group). Looking at references, it seems "empty" is the preferred term for category 0 (and terminal preferred over trivial for category 1)
https://ncatlab.org/nlab/show/terminal+category
https://ncatlab.org/nlab/show/empty+category
https://en.wikipedia.org/wiki/Category_of_small_categories
So I think `Empty` is better.

I am considering renaming `EmptyCategory` to `EmptyCategoryStyle` and `EmptyCategoryProduct` to `EmptyCategory` (as we may decide the empty category not to be a `CategoyProduct`)

Reply:
Thanks for the clarification, I think I had the terminology a bit off, I meant to suggest `TrivialSector`, i.e. an object that acts as a universal trivial sector for any group, so if you fuse it with any other sector/space you get that sector/space back. Does that coincide with the terminology "empty category"?

I think maybe part of the confusion is also that we are conflating the terminology category/group with sector in the code right now, i.e. our `AbstractCategory`s, when they are groups, are really symmetry sectors since they have sector labels, which is why I was thinking in terms of the terminology "trivial sector".

I do think we should consider if we need so many objects that act as "empty category"/"trivial sector". I was hoping that we could avoid the need for defining `EmptyCategoryProduct`, say if we made other code more generic so that it automatically handles the empty case. Additionally, as I think you are suggesting, I would wonder about the need for explicitly having an `EmptyCategory` definition, though I would also hope that maybe we can avoid having an `EmptyCategoryStyle` and again rely on generic code that implicitly handles the empty category product limit.

@emstoudenmire your input on this issue discussion would be helpful.

Reply:
The initial designed used a trait to precisely specify the kind of fusion category. It turned out that what we really need is a way to distinguish the very specific abelian group behavior from the generic one. I now defined just 2 traits, `AbelianStyle` and `NotAbelianStyle`. The empty category case falls into the `AbelianStyle`.

Reply:
Great, glad to see a reduction in the number of types/traits, the code definitely looks simpler.